### PR TITLE
Simplify debian packages.

### LINKED ISSUE #21698: Setting password during installation
I installed a CH from debian repos (just copied from docs) and installation process asking me for a password, I typed my testing password ended by "();" witouth quotes.

After install I can access without any password.

users.xml show like "<password></password>" for default user.

¯\\\_(ツ)\_/¯

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Simplify debian packages. This fixes #21698.

### CODE REVIEW

File: debian/clickhouse.limits [hunk 1]
```diff
@@ -1,2 +0,0 @@
-clickhouse	soft	nofile	262144
```
Comment (filimonov) [inline debian/clickhouse.limits:1]: Why removed?

Comment (filimonov) [reply debian/clickhouse.limits:1]: Is it also embedded into binary?

Comment (alexey-milovidov) [reply debian/clickhouse.limits:1]: It's created by `Install.cpp`
Actually everything is created by `Install.cpp`, everything else is a cruft.

**Review by alesapin — APPROVED** [review]
Fewer Debian scripts and everything continues to work!

**Comment by alexey-milovidov** [comment]
Before this change, when user is installed .deb package interactively, it will ask about default password with pink screen, but:
- it works only in debian packages, not in rpm;
- it works only during interactive installation, not in scripted install;
- for some reason, it was completely broken and confused people (but not always, I remember that sometimes it worked).

Our Install.cpp script is also asking for default password during installation. Also only in interactive more (otherwise it does not make sense). But if debian package is installed in interactive mode, looks like it does not asking for password (`dpkg -i` runs subprocesses without access to terminal?). It needs to be validated manually.

Also creation of cron files and limits files has to be validated manually.

**Comment by alexey-milovidov** [comment]
https://youtu.be/BBM0QbpPA0Q?t=640 (run at 2x speed)

**Comment by alexey-milovidov** [comment]
Cron files and limits are successfully created.
But it did not ask for default password during installation:

```
milovidov@milovidov-desktop:~/work$ mkdir packages
milovidov@milovidov-desktop:~/work$ cd packages
milovidov@milovidov-desktop:~/work/packages$ wget https://clickhouse-builds.s3.yandex.net/22976/5a5590580757550253a99fa50a51c6bb0af2c065/clickhouse_build_check/clang-11_relwithdebuginfo_none_bundled_unsplitted_disable_False_deb/clickhouse-server_21.5.1.6533_all.deb https://clickhouse-builds.s3.yandex.net/22976/5a5590580757550253a99fa50a51c6bb0af2c065/clickhouse_build_check/clang-11_relwithdebuginfo_none_bundled_unsplitted_disable_False_deb/clickhouse-common-static_21.5.1.6533_amd64.deb
--2021-04-13 20:24:53--  https://clickhouse-builds.s3.yandex.net/22976/5a5590580757550253a99fa50a51c6bb0af2c065/clickhouse_build_check/clang-11_relwithdebuginfo_none_bundled_unsplitted_disable_False_deb/clickhouse-server_21.5.1.6533_all.deb
Resolving clickhouse-builds.s3.yandex.net (clickhouse-builds.s3.yandex.net)... 2a02:6b8::2:158, 93.158.134.158
Connecting to clickhouse-builds.s3.yandex.net (clickhouse-builds.s3.yandex.net)|2a02:6b8::2:158|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 75118 (73K) [application/octet-stream]
Saving to: ‘clickhouse-server_21.5.1.6533_all.deb’

clickhouse-server_21.5.1.6533_all.deb                         100%[===============================================================================================================================================>]  73,36K  --.-KB/s    in 0,04s   

2021-04-13 20:24:53 (1,95 MB/s) - ‘clickhouse-server_21.5.1.6533_all.deb’ saved [75118/75118]

--2021-04-13 20:24:53--  https://clickhouse-builds.s3.yandex.net/22976/5a5590580757550253a99fa50a51c6bb0af2c065/clickhouse_build_check/clang-11_relwithdebuginfo_none_bundled_unsplitted_disable_False_deb/clickhouse-common-static_21.5.1.6533_amd64.deb
Reusing existing connection to [clickhouse-builds.s3.yandex.net]:443.
HTTP request sent, awaiting response... 200 OK
Length: 164943504 (157M) [application/octet-stream]
Saving to: ‘clickhouse-common-static_21.5.1.6533_amd64.deb’

clickhouse-common-static_21.5.1.6533_amd64.deb                100%[===============================================================================================================================================>] 157,30M  21,2MB/s    in 6,5s    

2021-04-13 20:25:00 (24,4 MB/s) - ‘clickhouse-common-static_21.5.1.6533_amd64.deb’ saved [164943504/164943504]

FINISHED --2021-04-13 20:25:00--
Total wall clock time: 6,8s
Downloaded: 2 files, 157M in 6,5s (24,2 MB/s)
milovidov@milovidov-desktop:~/work/packages$ docker run -it --rm --volume $(pwd):/packages ubuntu:focal /bin/bash
root@41e31e4ea029:/# cd /packages/
root@41e31e4ea029:/packages# dpkg -i clickhouse-server_21.5.1.6533_all.deb clickhouse-common-static_21.5.1.6533_amd64.deb
Selecting previously unselected package clickhouse-server.
(Reading database ... 4121 files and directories currently installed.)
Preparing to unpack clickhouse-server_21.5.1.6533_all.deb ...
Unpacking clickhouse-server (21.5.1.6533) ...
Selecting previously unselected package clickhouse-common-static.
Preparing to unpack clickhouse-common-static_21.5.1.6533_amd64.deb ...
Unpacking clickhouse-common-static (21.5.1.6533) ...
Setting up clickhouse-common-static (21.5.1.6533) ...
Setting up clickhouse-server (21.5.1.6533) ...
debconf: unable to initialize frontend: Dialog
debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (Can't locate Term/ReadLine.pm in @INC (you may need to install the Term::ReadLine module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.30.0 /usr/local/share/perl/5.30.0 /usr/lib/x86_64-linux-gnu/perl5/5.30 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.30 /usr/share/perl/5.30 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base) at /usr/share/perl5/Debconf/FrontEnd/Readline.pm line 7.)
debconf: falling back to frontend: Teletype
ClickHouse binary is already located at /usr/bin/clickhouse
Symlink /usr/bin/clickhouse-server already exists but it points to /clickhouse. Will replace the old symlink to /usr/bin/clickhouse.
Creating symlink /usr/bin/clickhouse-server to /usr/bin/clickhouse.
Creating symlink /usr/bin/clickhouse-client to /usr/bin/clickhouse.
Creating symlink /usr/bin/clickhouse-local to /usr/bin/clickhouse.
Creating symlink /usr/bin/clickhouse-benchmark to /usr/bin/clickhouse.
Symlink /usr/bin/clickhouse-copier already exists but it points to /clickhouse. Will replace the old symlink to /usr/bin/clickhouse.
Creating symlink /usr/bin/clickhouse-copier to /usr/bin/clickhouse.
Creating symlink /usr/bin/clickhouse-obfuscator to /usr/bin/clickhouse.
Creating symlink /usr/bin/clickhouse-git-import to /usr/bin/clickhouse.
Creating symlink /usr/bin/clickhouse-compressor to /usr/bin/clickhouse.
Creating symlink /usr/bin/clickhouse-format to /usr/bin/clickhouse.
Symlink /usr/bin/clickhouse-extract-from-config already exists but it points to /clickhouse. Will replace the old symlink to /usr/bin/clickhouse.
Creating symlink /usr/bin/clickhouse-extract-from-config to /usr/bin/clickhouse.
Creating clickhouse group if it does not exist.
 groupadd -r clickhouse
Creating clickhouse user if it does not exist.
 useradd -r --shell /bin/false --home-dir /nonexistent -g clickhouse clickhouse
Will set ulimits for clickhouse user in /etc/security/limits.d/clickhouse.conf.
Creating config directory /etc/clickhouse-server/config.d that is used for tweaks of main server configuration.
Creating config directory /etc/clickhouse-server/users.d that is used for tweaks of users configuration.
Config file /etc/clickhouse-server/config.xml already exists, will keep it and extract path info from it.
/etc/clickhouse-server/config.xml has /var/lib/clickhouse/ as data path.
/etc/clickhouse-server/config.xml has /var/log/clickhouse-server/ as log path.
Users config file /etc/clickhouse-server/users.xml already exists, will keep it and extract users info from it.
 chown --recursive clickhouse:clickhouse '/etc/clickhouse-server'
Creating log directory /var/log/clickhouse-server/.
Creating data directory /var/lib/clickhouse/.
Creating pid directory /var/run/clickhouse-server.
 chown --recursive clickhouse:clickhouse '/var/log/clickhouse-server/'
 chown --recursive clickhouse:clickhouse '/var/run/clickhouse-server'
 chown clickhouse:clickhouse '/var/lib/clickhouse/'
Password for default user is empty string. See /etc/clickhouse-server/users.xml and /etc/clickhouse-server/users.d to change it.
Setting capabilities for clickhouse binary. This is optional.
Cannot set 'net_admin' or 'ipc_lock' or 'sys_nice' capability for clickhouse binary. This is optional. Taskstats accounting will be disabled. To enable taskstats accounting you may add the required capability later manually.

ClickHouse has been successfully installed.

Start clickhouse-server with:
 sudo clickhouse start

Start clickhouse-client with:
 clickhouse-client

root@41e31e4ea029:/packages# ls -l /etc/security/
access.conf     group.conf      limits.conf     limits.d/       namespace.conf  namespace.d/    namespace.init  opasswd         pam_env.conf    sepermit.conf   time.conf       
root@41e31e4ea029:/packages# cat /etc/security/limits.d/clickhouse.conf 
clickhouse      soft    nofile  262144
clickhouse      hard    nofile  262144
root@41e31e4ea029:/packages# cat /etc/cron.d/                           
clickhouse-server  e2scrub_all        
root@41e31e4ea029:/packages# cat /etc/cron.d/clickhouse-server 
#*/10 * * * * root (which service > /dev/null 2>&1 && (service clickhouse-server condstart ||:)) || /etc/init.d/clickhouse-server condstart > /dev/null 2>&1
```

**Comment by alexey-milovidov** [comment]
Ok. Now it is asking for password.

But how to check that it will not ask when installing in "headless" mode?

**Comment by alexey-milovidov** [comment]
`clickhouse install > tmp.txt 2>&1`

Looks alright.

**Comment by alexey-milovidov** [comment]
`dpkg -i * > tmp.txt 2>&1`

Alright.

**Comment by filimonov** [comment]
> Ok. Now it is asking for password.
> 
> But how to check that it will not ask when installing in "headless" mode?

Check also:

```
env DEBIAN_FRONTEND=noninteractive \
        apt-get install  --yes ... \
```

**Comment by alexey-milovidov** [comment]
Unfortunately it keeps asking password even if the variable `DEBIAN_FRONTEND=noninteractive` is set.
(at least when terminal is available)

I can add a check for this variable in `Install.cpp`.

**Comment by alexey-milovidov** [comment]
Now it's Ok:
```
wget https://clickhouse-builds.s3.yandex.net/22976/2ce494aa0fa9f9fd631c28340c443db7ad771eea/clickhouse_build_check/clang-11_relwithdebuginfo_none_bundled_unsplitted_disable_False_deb/clickhouse-server_21.5.1.6566_all.deb https://clickhouse-builds.s3.yandex.net/22976/2ce494aa0fa9f9fd631c28340c443db7ad771eea/clickhouse_build_check/clang-11_relwithdebuginfo_none_bundled_unsplitted_disable_False_deb/clickhouse-common-static_21.5.1.6566_amd64.deb

docker run -it --rm --volume $(pwd):/packages ubuntu:focal /bin/bash

env DEBIAN_FRONTEND=noninteractive apt-get install --yes ./packages/*.deb
```

**Comment by filimonov** [comment]
> Unfortunately it keeps asking password even if the variable `DEBIAN_FRONTEND=noninteractive` is set.
> (at least when terminal is available)
> 
> I can add a check for this variable in `Install.cpp`.

Personally i wou

> Unfortunately it keeps asking password even if the variable `DEBIAN_FRONTEND=noninteractive` is set.
> (at least when terminal is available)
> 
> I can add a check for this variable in `Install.cpp`.

I don't think the solution is safe. Fontend can be be also kde / gnome / web etc. I don't think they will do proper redirects if stdin / stdout.

**Comment by filimonov** [comment]
IMHO when we are executed inside deb we should only use deb standard ways to interact with user (or not interact with him at all).

**Comment by alexey-milovidov** [comment]
> Fontend can be be also kde / gnome / web etc. I don't think they will do proper redirects of stdin / stdout.

Let's check it - it should not be able to use tty.

**Comment by alexey-milovidov** [comment]
> IMHO when we are executed inside deb we should only use deb standard ways to interact with user (or not interact with him at all).

I'm too tired of Debian ways to do the things as well as RedHat ways to do the same things.

**Comment by filimonov** [comment]
> > IMHO when we are executed inside deb we should only use deb standard ways to interact with user (or not interact with him at all).
> 
> I'm too tired of Debian ways to do the things as well as RedHat ways to do the same things.

Interacting with user during package installation sucks. In redhat there is no 'official' way for that at all.

**Comment by filimonov** [comment]
For example i can start typing next command (or it's waiting in input buffer because i did copy & paste few commands). That password prompt will eat it accidebtally etc.

**Comment by filimonov** [comment]
> For example i can start typing next command (or it's waiting in input buffer because i did copy & paste few commands). That password prompt will eat it accidebtally etc.

Did it twice already... :|

**Pull request merged**